File: AD-HOC/2187.py
entrada = int(input())
t = 0
while entrada != 0:
    cinquenta = entrada//50
    dez = entrada % 50//10
    cinco = entrada % 50 % 10//5
    um = entrada % 50 % 10 % 5

    t += 1
    print('Teste %d'%t)
    print('%d %d %d %d'%(cinquenta, dez, cinco, um))
    print()

    entrada = int(input())

File: AD-HOC/2187.c
#include <stdio.h>
#include <stdlib.h>

int main(){
    int entrada = 0, t = 0, cinq = 0, dez = 0, cinc = 0, um = 0;
    scanf("%d", &entrada);
    while(entrada != 0){
       cinq = (int)entrada / 50;
       dez = (int)entrada % 50 / 10;
       cinc = (int)entrada % 50 % 10 / 5;
       um = (int)entrada % 50 % 10 % 5;

       t += 1;
       printf("Teste %d\n", t);
       printf("%d %d %d %d\n\n", cinq, dez, cinc, um);
       scanf("%d", &entrada);
    }

    return 0;
}
